rosbridge_library/src/rosbridge_library/internal/message_conversion.py

```diff
@@ -420,10 +489,41 @@ def _to_object_inst(msg, rostype, roottype, clock, inst, stack):
         field_rostype = inst_fields[field_name]
         field_inst = getattr(inst, field_name)
 
-        field_value = _to_inst(
-            msg[field_name], field_rostype, roottype, clock, field_inst, field_stack
+        field_inst_value = _to_inst(
+            field_value, field_rostype, roottype, clock, field_inst, field_stack
         )
 
-        setattr(inst, field_name, field_value)
+        setattr(inst, field_name, field_inst_value)
 
     return inst
+
+
+def _remove_list_indicators(rostype: str) -> str:
+    """
+    Remove list indicators from rostype string.
+
+    Handles patterns like:
+    - sequence<double, 3> -> double
+    - sequence<string> -> string
+    - uint8[] -> uint8
+    - char[10] -> char
+
+    :param rostype: The ROS type string that may contain list indicators
+    :type rostype: str
+    :return: The base type without list indicators
+    :rtype: str
+    :raises ValueError: If the rostype doesn't match expected list patterns
+    """
+    # Try sequence pattern first: sequence<type, length> or sequence<type>
+    list_match = re.search(list_tokens, rostype)
+    if list_match:
+        return list_match.group(1)
+
+    # Try bounded array pattern: type[length] or type[]
+    bounded_match = re.search(bounded_array_tokens, rostype)
+    if bounded_match:
+        return bounded_match.group(1)
+
+    # If neither pattern matches, this isn't a valid list type
+    err_msg = f"Invalid rostype for list: {rostype}"
+    raise ValueError(err_msg)
```

Comment:
```suggestion
    msg = f"Invalid rostype for list: {rostype}"
    raise ValueError(msg)
```

Reply:
`msg` variable name is used throughout this module for JSON messages, I used `err_msg` for error messages to differentiate between the two and make static type checker happy